### PR TITLE
4pi beam convolution support (conviqt) for satellite pipeline

### PR DESCRIPTION
**Pull request by zonca**
Not ready for review yet, just added to more easily ask for feedback

### CODE REVIEW

**Comment by zonca** [comment]
it currently runs to completion on a simple test case, still figuring out if the output is reasonable

**Comment by zonca** [comment]
testing in https://github.com/zonca/pico-simulations/issues/17

File: pipelines/toast_satellite_sim.py [hunk 1]
```diff
@@ -243,6 +296,32 @@ def main():
                         help='Solar system speed galactic longitude[degrees]',
                         type=float, default=263.99)
 
+    # Conviqt beam convolution
+
+    parser.add_argument( "--run_conviqt", required=False, default=False,
+        action="store_true", help="Convolve map with full beam alms" )
+    parser.add_argument( "--conviqt_beamfile_alm", required=False, default="beam_DETECTOR_alm.fits",
```
Comment (zonca) [inline pipelines/toast_satellite_sim.py]: should we instead have the path to the beam `alms` inside the focalplane pkl?

File: pipelines/toast_satellite_sim.py [hunk 2]
```diff
@@ -243,6 +296,32 @@ def main():
...
+        help="Full beam alms filename, they should be defined in the same reference frame"
+        "both for A and B detectors, i.e. no rotation, not even for the polarization angle is"
+        "performed by TOAST."
```
Comment (zonca) [inline pipelines/toast_satellite_sim.py]: @keskitalo about the reference frame, I would start now supporting just 1. which is currently used by Karl for PICO that defines all the beams in the same reference frame, see https://github.com/zonca/pico-simulations/issues/17

File: pipelines/toast_satellite_sim.py [hunk 3]
```diff
@@ -124,6 +138,45 @@ def simulate_sky_signal(args, comm, data, mem_counter, focalplanes, subnpix, loc
...
+    local_dets = extract_local_dets(data)
+
+    for pattern in args.conviqt_beamfile_alm.split(','):
+        detectordata = []
+        for det in local_dets:
+            skyfile = args.conviqt_input_alm.replace('DETECTOR', det)
+            beamfile = pattern.replace('DETECTOR', det)
+            # polarization leakage
+            epsilon = 0.
+            psipol = 0. if det.endswith("A") else np.pi/2
```
Comment (zonca) [inline pipelines/toast_satellite_sim.py]: @keskitalo assuming `psipol` 90 deg is just temporary. should we add this to the focalplane pkl?

Comment (zonca) [reply pipelines/toast_satellite_sim.py]: or computing it from the detector quaternion?

**Comment by keskitalo** [comment]
This PR is now superseded by #304

**Pull request closed without merging**